python/tests/io/test_s3fs.py

```diff
@@ -0,0 +1,194 @@
+    output_file = s3fs_fileio.new_output(location=f"s3://testbucket/{filename}")
+    f = output_file.create()
+    f.write(b"foo")
+    f.close()
+
+    input_file = s3fs_fileio.new_input(f"s3://testbucket/{filename}")
+    assert input_file.open().read() == b"foo"
+
+    s3fs_fileio.delete(input_file)
+
+
+@pytest.mark.s3
+def test_getting_length_of_file(s3fs_fileio):
+    """Test getting the length of an S3fsInputFile and S3fsOutputFile"""
+    filename = str(uuid.uuid4())
+
+    output_file = s3fs_fileio.new_output(location=f"s3://testbucket/{filename}")
+    f = output_file.create()
+    f.write(b"foobar")
+    f.close()
+    assert len(output_file) == 6
+
+    input_file = s3fs_fileio.new_input(location=f"s3://testbucket/{filename}")
+    assert len(input_file) == 6
+
+    s3fs_fileio.delete(output_file)
+
+
+@pytest.mark.s3
+def test_s3fs_tell(s3fs_fileio):
+    """Test finding cursor position for an S3fsInputFile"""
+
+    filename = str(uuid.uuid4())
+
+    output_file = s3fs_fileio.new_output(location=f"s3://testbucket/{filename}")
+    f = output_file.create()
+    f.write(b"foobar")
+    f.close()
+
+    input_file = s3fs_fileio.new_input(location=f"s3://testbucket/{filename}")
+    f = input_file.open()
+
+    f.seek(0)
+    assert f.tell() == 0
+    f.seek(1)
+    assert f.tell() == 1
+    f.seek(3)
+    assert f.tell() == 3
+    f.seek(0)
+    assert f.tell() == 0
+
+
+@pytest.mark.s3
+def test_s3fs_read_specified_bytes(s3fs_fileio):
+    """Test reading a specified number of bytes from an S3fsInputFile"""
+
+    filename = str(uuid.uuid4())
+    output_file = s3fs_fileio.new_output(location=f"s3://testbucket/{filename}")
+    f = output_file.create()
+    f.write(b"foo")
+    f.close()
+
+    input_file = s3fs_fileio.new_input(location=f"s3://testbucket/{filename}")
+    f = input_file.open()
+
+    f.seek(0)
+    assert b"f" == f.read(size=1)
+    f.seek(0)
+    assert b"fo" == f.read(size=2)
+    f.seek(1)
+    assert b"o" == f.read(size=1)
+    f.seek(1)
+    assert b"oo" == f.read(size=2)
+    f.seek(0)
+    assert b"foo" == f.read(size=999)  # test reading amount larger than entire content length
+
+    s3fs_fileio.delete(input_file)
+
+
+@pytest.mark.s3
+def test_raise_on_opening_an_s3_file_not_found(s3fs_fileio):
+    """Test that an S3fsInputFile raises appropriately when the file is not found"""
+
+    filename = str(uuid.uuid4())
+    input_file = s3fs_fileio.new_input(location=f"s3://testbucket/{filename}")
+    with pytest.raises(FileNotFoundError) as exc_info:
+        input_file.open().read()
+
+    assert filename in str(exc_info.value)
+
+
+@pytest.mark.s3
+def test_checking_if_a_file_exists(s3fs_fileio):
+    """Test checking if a file exists"""
+
+    non_existent_file = s3fs_fileio.new_input(location="s3://testbucket/does-not-exist.txt")
+    assert not non_existent_file.exists()
+
+    filename = str(uuid.uuid4())
+    output_file = s3fs_fileio.new_output(location=f"s3://testbucket/{filename}")
```

Comment:
nit: maybe we can also assert `output_file` does not exist before it get created?

Reply:
Fixed! Thanks @dramaticlly